kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/utils/MessageMetadataUtils.java

```diff
@@ -191,6 +204,12 @@ public static CompletableFuture<Position> asyncFindPosition(final ManagedLedger
             } finally {
                 entry.release();
             }
-        });
+        }
+
+        @Override
+        public String toString() {
+            return "FindEntryByOffset{ " + offset + "}";
+        }
```

Comment:
Good find! I think we can add the info of the managed ledger like

```java
            return "FindEntryByOffset{ managed ledger: " + managedLedger.getName() + ", offset: " + offset + " }";
```

And we can only reference the name when constructing the predicate.

```java
        return managedLedger.asyncFindPosition(new FindEntryByOffset(managedLedger.getName(),
                offset, skipMessagesWithoutIndex));
    }

    @AllArgsConstructor
    private static class FindEntryByOffset implements Predicate<Entry> {

        private final String managedLedgerName;
```

Reply:
IIRC the topic name is already reported in the logs of asyncFindPosition:

https://github.com/apache/pulsar/blob/master/managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java#L1710